app/models/course.rb

```diff
@@ -265,9 +265,34 @@ def eval_category
     end
   end
 
+  def self.search(term_id, status, title_parts, manager_parts)
+    @candidates = Course.all.limit(COURSE_SEARCH_MAX_SIZE)
+    @candidates = @candidates.where(term_id: term_id) unless term_id.empty?
+    @candidates = @candidates.where(status: status) unless status.empty?
+    @candidates = @candidates.where('title like ?', '%' + title_parts + '%') if title_parts.present?
+    if manager_parts.present?
+      manager_parts = manager_parts.tr('　',' ')
+      search_words = manager_parts.split(' ');
+      case search_words.length
+      when 2
+        family_name = search_words[0]
+        given_name = search_words[1]
+        logical_operation = 'and'
+      else
+        family_name = given_name = manager_parts
+        logical_operation = 'or'
+      end
+      manager_ids = User.where('family_name like ? ' + logical_operation + ' given_name like ? ', '%' + family_name + '%', '%' + given_name + '%').pluck(:id)
+      course_ids = CourseMember.where(role: 'manager').where('user_id IN (?)', manager_ids).pluck(:course_id).uniq
+      @candidates = @candidates.where('id IN(?)', course_ids)
+    end
+    @candidates
```

Comment:
Here, limit the candidates number.

`@candidates.limit(COURSE_SEARCH_MAX_SIZE)`

Reply:
:question: May I ask a question?
The  sql I executed searching was as follows in the current source code.
-- 
SELECT  "courses".* FROM "courses" WHERE "courses"."term_id" = ? AND "courses"."status" = ? LIMIT ?  [["term_id", 1], ["status", "open"], ["LIMIT", 50]]
--
It seems "Rails" automatically assemble the efficient sql.
Is this depend on the environment?

Reply:
This was my misunderstanding.
The active record query methods are executed for the first time when those are needed.
So in this case, the same result will be obtained regardless of the execution place of the limit methods.

But I think it is easier to understand if limit is listed after where.